raiden_libs/network/matrix/client.py

```diff
@@ -36,9 +39,12 @@ def __init__(
         )
         self.should_listen = False
         self.sync_thread = None
-        self.greenlets = list()  # type: List[gevent.Greenlet]
+        self.greenlets: List[gevent.Greenlet] = list()
         self.api.session.headers.update({'Connection': 'close'})
 
+        # locks each account_data's key until it's _sync'ed
+        self.account_data_locks: Dict[str, gevent.lock.Semaphore] = dict()
```

Comment:
Wouldn't it make more sense to use `gevent.Event()` here? Seems to me like you're already using the semaphores like events already.

Also using a `defaultdict` could make the `set_|get_account_data()` a bit easier.

Reply:
I thought about it, but that may not work or be less expresive, as the intent here is to make `set_account_data` block and wait until we get the same data we set on the next `sync` loop. Using event can make this notification happen, but won't help blocking in the case a concurrent call to `set_account_data` was made, which is the main purpose of this locking mechanism.

Reply:
Notice also that the `set_account_data` should trigger an event immediatelly, so the `sync` loop should happen right after (not in 30s only or so), so it's pretty quick and just a race condition safe net.